public/pages/Correlations/containers/CreateCorrelationRule.tsx

```diff
@@ -234,20 +234,20 @@ export const CreateCorrelationRule: React.FC<CreateCorrelationRuleProps> = (
       setNotificationChannels(parsedChannels);
       setLoadingNotifications(false);
     };
+    const setInitialRuleValues = async (ruleId: string) => {
+      const ruleRes = await correlationStore.getCorrelationRule(ruleId);
+      if (ruleRes) {
+        setInitialValues(ruleRes);
+      }
+    };
+    
     if (props.history.location.state?.rule) {
       setAction('Edit');
-      setInitialValues(props.history.location.state?.rule);
+      setInitialRuleValues(props.history.location.state?.rule.id);
```

Comment:
why do we need to fetch rule again in this case? shouldn't the passed in rule have the triggers as well?

Reply:
addressed